app/models/observation.rb

```diff
@@ -1218,6 +1218,32 @@ def collector_and_number
     collection_numbers.first.format_name
   end
 
+  ##############################################################################
+  #
+  #  :section: Sources
+  #
+  ##############################################################################
+
+  enum source:
+        {
+          mo_website: 1,
+          mo_android_app: 2,
+          mo_iphone_app: 3,
+          mo_api: 4
+        }
+
+  # Which agent created this observation?
+  def self.all_sources
+    [:mo_website, :mo_android_app, :mo_iphone_app, :mo_api]
+  end
```

Comment:
Should this return strings instead of symbols?
If symbols, maybe the following to DRY things
```suggestion
  def self.all_sources
    self.sources.keys.map(&:to_sym)
  end
```
or maybe just define a frozen constant.
(And analogous suggestions if strings are needed.)

Reply:
Thank you, good suggestion.  Taken care of, I think.